pkg/operator/controller/ingress/controller.go

```diff
@@ -409,7 +409,7 @@ func setDefaultDomain(ic *operatorv1.IngressController, ingressConfig *configv1.
 	return false
 }
 
-func setDefaultPublishingStrategy(ic *operatorv1.IngressController, platformStatus *configv1.PlatformStatus, domainMatchesBaseDomain bool) bool {
+func setDefaultPublishingStrategy(ic *operatorv1.IngressController, platformStatus *configv1.PlatformStatus, domainMatchesBaseDomain bool, ingressConfig *configv1.Ingress) bool {
```

Comment:
I think some of the refactoring in `setDefaultPublishingStrategy` is unnecessary, but you're also missing some defaulting logic for the case `ic.Status.EndpointPublishingStrategy` is nil.  I'll submit a PR rather than try to explain in a bunch of comments.